platform.md

```diff
@@ -969,6 +1028,7 @@ Where:
   [[entries.providers]]
     id = "<buildpack ID>"
     version = "buildpack Version"
+    extension = false
```

Comment:
> is it possible to split extensions into its own table array

Due to the way extensions interoperate with buildpacks in "resolving" a build plan, this would significantly complicate the implementation :( - but I'm happy to make the clarification changes proposed above.

Reply:
In future phases do we see this file being read by the platform in order to apply the Dockerfiles (ie. the docker daemon case)? If so, the platform doesn't care about buildpacks and I would further push that we take on that complexity at the lifecycle instead of making _every_ platforms iterate through entries that don't apply to them.

Reply:
Luckily for the lifecycle, I don't think platforms would ever need to read this file. It is filtered by the lifecycle during the generate and build phases. At the point where platforms need to care about extensions, all they need to consume is Dockerfiles (and args).

Reply:
Hmmm... that brings up more questions. Those questions are likely to be for future phases but I think they may still apply to this conversation. This PR doesn't talk about what the platform would do with those Dockerfile (IIRC, the lifecycle is going to manage that in phase 1). In the next phase, wouldn't the platform use this `plan.toml` to know the order in which the docker files should be applied? Also, mostly out of curiosity, would it include the args needed to execute the Dockerfiles?

Reply:
Re: the question of args - I think this will necessitate the bringing back of something like `generated.toml` to collate those in one accessible place. `group.toml` can be used to determine the order of applying the Dockerfiles.

Reply:
@jromero I made some changes in e8dda22 - WDYT?

Reply:
I think we need both. Looking at the existing files, I don't see a place for build args to go:

* `analyzed.toml` - output of analyze phase, contains image metadata
* `order.toml` - input to detect phase
* `group.toml` - (this should really be called `detected.toml`) output of detect phase, contains the extensions and buildpacks that detected
* `plan.toml` - output of detect phase, contains the dependencies that each extension or buildpack is expected to provide, **input to generate / build phases**
* `metadata.toml` - (this should really be called `built.toml`) output of build phase, collates information output by buildpacks (such as process types)
* `report.toml` - (this should really be called `exported.toml`) output of export phase

The file being proposed:
* `generated.toml` - **output of generate phase**, collates information output by extensions (such as build args), input to extend phase

I wouldn't put build args in `plan.toml` because they are an output of generate whereas `plan.toml` is an input to generate.

To me it makes sense for extensions and buildpacks to use the same `plan.toml` because they collaborate on the plan - extensions "provide" dependencies that buildpacks "require".

However having this discussion has made me realize that perhaps we are missing yet another file - I will add as a separate comment since it is slightly orthogonal to the current discussion.

Reply:
New comment: https://github.com/buildpacks/spec/pull/308/files#r920191777